search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserFilterTransformer.java

```diff
@@ -17,14 +18,16 @@
 import io.camunda.search.clients.query.SearchQuery;
 import io.camunda.search.filter.UserFilter;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class UserFilterTransformer implements FilterTransformer<UserFilter> {
 
   @Override
   public SearchQuery toSearchQuery(final UserFilter filter) {
 
     return and(
-        filter.key() == null ? null : term(KEY, filter.key()),
+        filter.keys() == null || filter.keys().isEmpty() ? null : orTerms(KEY, filter.keys()),
```

Comment:
We can use a `terms` filter here: `SearchQueryBuilders.terms().field(KEY).longTerms(filter.keys()).build()`